src/components/dialogs/parameters/network-visualizations/network-visualizations-utils.ts

```diff
@@ -0,0 +1,153 @@
+    lineFlowColorMode: LineFlowColorMode;
+    lineFlowAlertThreshold: number;
+    mapManualRefresh: boolean;
+    mapBaseMap: string;
+    diagonalLabel: boolean;
+    centerLabel: boolean;
+    substationLayout: string;
+    componentLibrary: unknown;
+    initNadWithGeoData: boolean;
+}
+export interface NestedObject {
+    [key: string]: NestedObject | string | number | boolean;
+}
+
+export const intlLineFlowModeOptions = () => [
+    {
+        id: LineFlowMode.STATIC_ARROWS,
+        label: 'StaticArrows',
+    },
+    {
+        id: LineFlowMode.ANIMATED_ARROWS,
+        label: 'AnimatedArrows',
+    },
+    {
+        id: LineFlowMode.FEEDERS,
+        label: 'Feeders',
+    },
+];
+
+export const intlLineFlowColorModeOptions = () => [
+    {
+        id: LineFlowColorMode.NOMINAL_VOLTAGE,
+        label: 'NominalVoltage',
+    },
+    {
+        id: LineFlowColorMode.OVERLOADS,
+        label: 'Overloads',
+    },
+];
+export const intlMapBaseMapOptions = () => [
+    {
+        id: MAP_BASEMAP_MAPBOX,
+        label: 'Mapbox',
+    },
+    {
+        id: MAP_BASEMAP_CARTO,
+        label: 'Carto',
+    },
+    {
+        id: MAP_BASEMAP_CARTO_NOLABEL,
+        label: 'CartoNoLabel',
+    },
+];
+
+export const fromNetworkVisualizationsParamsDataToFormValues = (parameters: NetworkVisualizationsParams) => {
+    return {
+        [TabValue.MAP]: {
+            [PARAM_LINE_FULL_PATH]: parameters.lineFullPath,
+            [PARAM_LINE_PARALLEL_PATH]: parameters.lineParallelPath,
+            [PARAM_LINE_FLOW_MODE]: parameters.lineFlowMode,
+            [PARAM_LINE_FLOW_COLOR_MODE]: parameters.lineFlowColorMode,
+            [PARAM_LINE_FLOW_ALERT_THRESHOLD]: parameters.lineFlowAlertThreshold,
+            [PARAM_MAP_MANUAL_REFRESH]: parameters.mapManualRefresh,
+            [PARAM_MAP_BASEMAP]: parameters.mapBaseMap,
+        },
+        [TabValue.SINGLE_LINE_DIAGRAM]: {
+            [PARAM_DIAGONAL_LABEL]: parameters.diagonalLabel,
+            [PARAM_CENTER_LABEL]: parameters.centerLabel,
+            [PARAM_SUBSTATION_LAYOUT]: parameters.substationLayout,
+            [PARAM_COMPONENT_LIBRARY]: parameters.componentLibrary,
+        },
+        [TabValue.NETWORK_AREA_DIAGRAM]: {
+            [PARAM_INIT_NAD_WITH_GEO_DATA]: parameters.initNadWithGeoData,
+        },
+    };
+};
+
+export const intlSubstationLayoutOptions = () => [
+    {
+        id: SubstationLayout.HORIZONTAL,
+        label: 'HorizontalSubstationLayout',
+    },
+    {
+        id: SubstationLayout.VERTICAL,
+        label: 'VerticalSubstationLayout',
+    },
+];
+
+export const formatParametersToSend = (
+    parameters: NetworkVisualizationParametersForm,
+    changedParameters: Partial<NetworkVisualizationParametersForm>
+) => {
+    const getFlattenedObject = (nestedObject: NestedObject) =>
+        Object.values(nestedObject)
+            .flatMap((obj) => Object.entries(obj))
+            .reduce((acc, [key, value]) => ({ ...acc, [key]: value }), {});
```

Comment:
Here if you are trying to have all the params in forme of a Record<string, string | boolean | number>, i think it should be like this : 
```suggestion
            const getFlattenedObject = (nestedObject: NestedObject): Record<string, string | number | boolean> => {
    return Object.entries(nestedObject).reduce((acc, [key, value]) => {
        if (typeof value === 'object') {
            const flattenedChild = getFlattenedObject(value as NestedObject);
            return { ...acc, ...flattenedChild };
        } else {
            return { ...acc, [key]: value };
        }
    }, {} as Record<string, string | number | boolean>);
};
```

Reply:
I really don't see why I need to change this. What seems to be the problem with the actual getFlattenedObject?